devicetypes/smartthings/inovelli-dimmer.src/inovelli-dimmer.groovy

```diff
@@ -0,0 +1,528 @@
+		zwaveEvent(encapsulatedCommand)
+	} else {
+		log.warn "Unable to extract Secure command from $cmd"
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.crc16encapv1.Crc16Encap cmd) {
+	def version = commandClassVersions[cmd.commandClass as Integer]
+	def ccObj = version ? zwave.commandClass(cmd.commandClass, version) : zwave.commandClass(cmd.commandClass)
+	def encapsulatedCommand = ccObj?.command(cmd.command)?.parse(cmd.data)
+	if (encapsulatedCommand) {
+		log.debug "Parsed Crc16Encap into: ${encapsulatedCommand}"
+		zwaveEvent(encapsulatedCommand)
+	} else {
+		log.warn "Unable to extract CRC16 command from $cmd"
+	}
+}
+
+private secEncap(physicalgraph.zwave.Command cmd) {
+	log.debug "encapsulating command using Secure Encapsulation, command: $cmd"
+	zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+}
+
+private crcEncap(physicalgraph.zwave.Command cmd) {
+	log.debug "encapsulating command using CRC16 Encapsulation, command: $cmd"
+	zwave.crc16EncapV1.crc16Encap().encapsulate(cmd).format()
+}
+
+private encap(cmd, endpoint = null) {
+	if (cmd) {
+		if (endpoint) {
+			cmd = zwave.multiChannelV3.multiChannelCmdEncap(destinationEndPoint: endpoint).encapsulate(cmd)
+		}
+
+		if (zwaveInfo.zw.endsWith("s")) {
+			zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+		} else {
+			cmd.format()
+		}
+	}
+}
+
+private encapSequence(cmds, Integer delay = 250) {
+	delayBetween(cmds.collect { encap(it) }, delay)
+}
+
+private getParameterMap() {
+	[
+		[
+			name           : "Dimming Speed", key: "dimmingSpeed", type: "range",
+			parameterNumber: 1, size: 1, defaultValue: 3,
+			range          : "1..100",
+			description    : "How fast or slow the light turns on when you hold the switch in seconds (ie: dimming from 10-20%, 80-60%, etc). Value 0 - Instant On. This parameter can be set without a HUB from the Configuration Button. Finally, if you are using a,dumb switch in a 3-Way setting, this parameter will not work if you manually press the dumb switch (it will only work if you press the smart switch)."
+		],
+		[
+			name           : "Power On State", key: "powerOnState", type: "range",
+			parameterNumber: 11, size: 1, defaultValue: 0,
+			range          : "0..101",
+			description    : "When power is restored, the switch reverts to either On, Off, or Last Level. Example of how the values work: 0 = Off, 1-100 = Specific % On, 101 = Returns to Level before Power Outage. This parameter can be set without a HUB from the Configuration Button."
+		],
+		[
+			name           : "LED Indicator Color", key: "ledIndicatorColor", type: "range",
+			parameterNumber: 13, size: 2, defaultValue: 170,
+			range          : "0..255",
+			description    : "This will set the default color of the LED Bar. This is calculated by using a hue color circle (Value / 255 * 360). See website for more info. This parameter can be set without a HUB from the Configuration Button."
```

Comment:
I think we may need to remove "see website for more info" since they'll probably assume we mean _our_ website

Also isn't this what we're setting with the child device? Since we're controlling it that way it might not make sense to be duplicating the functionality here with a preference.

Reply:
Done